lib/streamer/cbor_streamer.go

```diff
@@ -0,0 +1,199 @@
+package streamer
+
+import (
+	"io"
+	"math"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/spacemonkeygo/errors"
+	"github.com/ugorji/go/codec"
+)
+
+var _ Mux = &CborMux{}
+
+type CborMux struct {
+	// the fd tip handles append; reads all use ReadAt and track their offsets individually
+	file  *os.File
+	codec *codec.Encoder
+	wmu   sync.Mutex
+}
+
+type cborMuxRow struct {
+	Label int    `json:"l"`
+	Msg   []byte `json:"m,omitempty"`
+	Sig   int    `json:"x,omitempty"` // 1->closed
+}
+
+func CborFileMux(filePath string) Mux {
+	file, err := os.OpenFile(filePath, os.O_CREATE|os.O_RDWR|os.O_EXCL, 0755)
+	if err != nil {
+		panic(errors.IOError.Wrap(err))
+	}
+	file.Write([]byte{codec.CborStreamArray})
```

Comment:
It looks like you've designed labels to independently drop in & out.
Given our use case, wouldn't we expect all labels to close around the same time?
Do you envision a case where we'll need this flexibility?

One implementation option: `CborMux.Appender(label int)` remembers the label added in a set that's kept on the CborMux and protected by the same mutex. Then `cborMuxAppender.Close()` would additionally call a new method `CborMux.CloseAppender(label int)` subtracting the label from the set.

The `CborMux` would emit a stream close header when `CloseAppender` is called and results in an empty set of open labels. Any attempts to write after that occurs would be an error.